File: parlib_soptpure/parlib/netimpl/network.py

import socket

from parlib.net.inetwork import INetwork
from parlib.netimpl.networkreader import NetworkReader

class Network(INetwork):
	def __init__(self):
		self.s = None
	
	def connect(self, host, port):
		try:
			self.s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
			self.s.connect((host, port))
			return True
		except socket.error:
			self.s.close()
			return False
		
	def sendMessage(self, message):
		try:
			n = self.s.sendall(message)
			#print "sent:", message
			#flush?
			return True
		except socket.error:
			return False
	
	def createNetworkReader(self):
		return NetworkReader(self.s)
	
	def setTimeout(self, timeoutmillis):
		self.s.settimeout(timeoutmillis / 1000.0)
		return True

File: experiments/proj_ftpclient/jclient/paskma/ftpclient/netimpl/Network.java
package paskma.ftpclient.netimpl;

import paskma.ftpclient.net.INetwork;
import paskma.ftpclient.net.INetworkReader;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.Socket;

/**
 * User: paskma
 * Date: Oct 10, 2011
 * Time: 12:45:10 PM
 */
public class Network implements INetwork {
	private Socket socket;

	public boolean connect(String host, int port) {
		try {
			socket = new Socket(host, port);
			return true;
		} catch (IOException e) {
			e.printStackTrace();
			return false;
		}
	}

	public INetworkReader createNetworkReader() {
		return new NetworkReader(socket);
	}

	public boolean sendMessage(String message) {
		try {
			socket.getOutputStream().write(message.getBytes());
			socket.getOutputStream().flush();
			System.out.println("sent: " + message);
			return true;
		} catch (IOException e) {
			return false;
		}
	}
}
